src/Compilers/Core/Portable/Symbols/ITypeSymbol.cs

```diff
@@ -168,6 +168,17 @@ public interface ITypeSymbol : INamespaceOrTypeSymbol
             NullableFlowState topLevelNullability,
             int position,
             SymbolDisplayFormat? format = null);
+
+        /// <summary>
+        /// Nullable annotation associated with the type, or <see cref="NullableAnnotation.None"/> if there are none.
+        /// </summary>
+        NullableAnnotation NullableAnnotation { get; }
```

Comment:
amazing! (seriously).  awesome job!

Reply:
just to check.  so there's no way to go from an ITypeSymbol to the nullability it was 'declared' with right?  @333fred @jasonmalinowski is htat ok?  is that something a feature needs?  what's the expectation on how they would get it?

i *think* but am not certain, that you should be ok with semantic model *depending* on what semantic model returns for ops.  For example, what does GetTypeInfo return?  does it return both "DeclaredType" and "Type"?  If so, great, if not... what is end consumption code expected to do?

Same with GetSymbolInfo does it return the symbol (say a field symbol) with teh current nullability at the current location, or does it return the symbol as it was declared?

I'm just trying to wrap my head around how someone is expected to consume things and what patterns will be more 'correct' and what will likely result in incorrect data/decision being made. 

Thanks!

Reply:
`GetTypeInfo` returns the current state, and `GetSymbolInfo` returns the declared information. The idea is that consumers will need to figure out what they need to know, and call the appropriate API, much like the differences between those APIs today.

Reply:
Interesting.  Note: i personally think that GetTypeInfo should be updated to include 'declared' info as well.  i think it's asking a *lot* to have to make people jump through these hoops.

In particular, GetSymbolInfo is not easy to use.  And hte relationship between the two and when/how people can expect the latter to work is not easy to understand and internalize.  

So just my 2c that it woudl be nice to have the api dedicated to understanding types at the semantic level be updated to understand this totally natural view of what a type is.  

Thanks! :)

Reply:
So when we moved the IDE to our existing wrappers, we didn't really run into this as a problem. We might have to switch a few places to now call GetSymbolInfo, but they're going to be places like where we're checking if we can assign something safely, and those could have (and perhaps should have) been using GetSymbolInfo.

(If anything, we "discovered" a few cases where GetTypeInfo worked that looked questionable to me in the first place, and switched the IDE to GetSymbolInfo for safety no matter what!)